packages/output/src/browser/output-contribution.ts

```diff
@@ -14,100 +14,168 @@
-            isVisible: widget => this.withWidget(widget, () => true),
-            execute: widget => this.withWidget(widget, outputWidget => this.clear(outputWidget))
+    registerCommands(registry: CommandRegistry): void {
+        super.registerCommands(registry);
+        registry.registerCommand(OutputCommands.CLEAR__WIDGET, {
+            isEnabled: () => this.withWidget(),
+            isVisible: () => this.withWidget(),
+            execute: () => this.widget.then(widget => widget.clear())
         });
-        commands.registerCommand(OutputCommands.SELECT_ALL, {
-            isEnabled: () => this.outputIsActiveContext.isEnabled(),
-            isVisible: () => this.outputIsActiveContext.isEnabled(),
-            execute: widget => this.withWidget(widget, outputWidget => outputWidget.selectAll())
+        registry.registerCommand(OutputCommands.LOCK__WIDGET, {
+            isEnabled: widget => this.withWidget(widget, output => !output.isLocked),
+            isVisible: widget => this.withWidget(widget, output => !output.isLocked),
+            execute: () => this.widget.then(widget => widget.lock())
+        });
+        registry.registerCommand(OutputCommands.UNLOCK__WIDGET, {
+            isEnabled: widget => this.withWidget(widget, output => output.isLocked),
+            isVisible: widget => this.withWidget(widget, output => output.isLocked),
+            execute: () => this.widget.then(widget => widget.unlock())
         });
     }
 
-    registerKeybindings(registry: KeybindingRegistry): void {
-        super.registerKeybindings(registry);
-        registry.registerKeybindings({
-            command: OutputCommands.SELECT_ALL.id,
-            keybinding: 'CtrlCmd+A',
-            context: OutputWidgetIsActiveContext.ID
+    registerMenus(registry: MenuModelRegistry): void {
+        super.registerMenus(registry);
+        registry.registerMenuAction(OutputContextMenu.TEXT_EDIT_GROUP, {
+            commandId: CommonCommands.COPY.id
+        });
+        registry.registerMenuAction(OutputContextMenu.COMMAND_GROUP, {
+            commandId: quickCommand.id,
+            label: 'Find Command...'
+        });
+        registry.registerMenuAction(OutputContextMenu.WIDGET_GROUP, {
+            commandId: OutputCommands.CLEAR__WIDGET.id,
+            label: 'Clear Output'
         });
     }
 
-    protected async clear(widget: OutputWidget): Promise<void> {
-        widget.clear();
+    canHandle(uri: URI): MaybePromise<number> {
+        return OutputUri.is(uri) ? 200 : 0;
     }
 
-    protected withWidget<T>(widget: Widget | undefined = this.tryGetWidget(), cb: (problems: OutputWidget) => T): T | false {
-        if (widget instanceof OutputWidget && widget.id === OUTPUT_WIDGET_KIND) {
-            return cb(widget);
-        }
-        return false;
+    open(uri: URI, options?: OpenerOptions): Promise<OutputWidget> {
+        return new Promise<OutputWidget>((resolve, reject) => {
+            if (!OutputUri.is(uri)) {
```

Comment:
why not to use `async/await`? this method will hang if something goes wrong in `openView` or  `setInptu`